_drafts/PR-blogs/planning-docs/example-code/pong/original/index.js

```diff
@@ -0,0 +1,123 @@
+let gameState = 'start';
+let paddle_1 = document.querySelector('.paddle_1');
+let paddle_2 = document.querySelector('.paddle_2');
+let board = document.querySelector('.board');
+let initial_ball = document.querySelector('.ball');
+let ball = document.querySelector('.ball');
+let score_1 = document.querySelector('.player_1_score');
+let score_2 = document.querySelector('.player_2_score');
+let message = document.querySelector('.message');
+let paddle_1_coord = paddle_1.getBoundingClientRect();
+let paddle_2_coord = paddle_2.getBoundingClientRect();
+let initial_ball_coord = ball.getBoundingClientRect();
+let ball_coord = initial_ball_coord;
+let board_coord = board.getBoundingClientRect();
+let paddle_common =
+	document.querySelector('.paddle').getBoundingClientRect();
+
+let dx = Math.floor(Math.random() * 4) + 3;
+let dy = Math.floor(Math.random() * 4) + 3;
+let dxd = Math.floor(Math.random() * 2);
+let dyd = Math.floor(Math.random() * 2);
+
+document.addEventListener('keydown', (e) => {
+if (e.key == 'Enter') {
+	gameState = gameState == 'start' ? 'play' : 'start';
+	if (gameState == 'play') {
+	message.innerHTML = 'Game Started';
+	message.style.left = 42 + 'vw';
+	requestAnimationFrame(() => {
+		dx = Math.floor(Math.random() * 4) + 3;
+		dy = Math.floor(Math.random() * 4) + 3;
+		dxd = Math.floor(Math.random() * 2);
+		dyd = Math.floor(Math.random() * 2);
+		moveBall(dx, dy, dxd, dyd);
+	});
+	}
+}
+if (gameState == 'play') {
+	if (e.key == 'w') {
```

Comment:
There's a lot of repeated code in here for moving paddles up and down, with only slight variations between the various implementations (which paddle we're moving, which direction, etc.). To make this code a little shorter, and to make it easier to understand at a glance, I'd suggest abstracting all this into `movePaddleDown` and `movePaddleUp` functions. That, combined with a little bit of thoughtful variable naming and some nice ES6 string-template code, can make this a lot easier to grok and work with: 

```js
// an example implementation for movePaddleUp
// movePaddleDown would be the same thing, but with all the math appropriately flipped around
const movePaddleUp = (paddle) => {
  const nextPosition = Math.max(
    board_coord.top,
    paddle.top - window.innerHeight * movementInterval
  );

  return `${nextPosition}px`;
};

// Example function calls: 
if (e.key == MovementKeys.P1_UP) {
  paddle_1.style.top = movePaddleUp(paddle_1_coord);
}

if (e.key == MovementKeys.P1_DOWN) {
  paddle_1.style.top = movePaddleDown(paddle_1_coord);
}
```